### PR TITLE
fix: Return proper http response codes for errors

### LINKED ISSUE #171: Return proper HTTP response codes for errors
Currently all cards return 200 (unless there is a server error, in which case 500).

It would be best to have specific codes for different types of errors to differentiate between them.

Example:

400 - Bad request

etc.

### PR DESCRIPTION
**Pull request by komen205**
## Description

Fixes #171

### Type of change

- [X] New feature (added a non-breaking change which adds functionality)

## How Has This Been Tested?

- [ ] Tested locally with a valid username
- [ ] Tested locally with an invalid username
- [ ] Ran tests with `composer test`
- [ ] Added or updated test cases to test new features

## Checklist:

- [X] I have checked to make sure no other [pull requests](https://github.com/DenverCoder1/github-readme-streak-stats/pulls?q=is%3Apr+sort%3Aupdated-desc+) are open for this issue
- [X] The code is properly formatted and is consistent with the existing code style
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by komen205** [comment]
@DenverCoder1 let me know if this works

File: src/card.php [hunk 1]
```diff
@@ -345,9 +345,10 @@ function convertSvgToPng(string $svg): string
  *
  * @param string|array $output The stats (array) or error message (string) to display
  */
-function renderOutput(string|array $output): void
+function renderOutput(string|array $output, int $respondeCode = 200): void
```
Comment (DenverCoder1) [inline src/card.php:348]:
```suggestion
function renderOutput(string|array $output, int $responseCode = 200): void
```

File: src/card.php [hunk 2]
```diff
@@ -345,9 +345,10 @@ function convertSvgToPng(string $svg): string
...
 {
     $requestedType = $_REQUEST['type'] ?? 'svg';
+    http_response_code($respondeCode);
```
Comment (DenverCoder1) [inline src/card.php:351]:
```suggestion
    http_response_code($responseCode);
```

**Review by DenverCoder1 — COMMENTED** [review]
The error codes also need to be set when throwing the exception

e.g. for errors that should return 404 response, 404 should be passed as the `$code` parameter.

```php
throw new InvalidArgumentException("Could not find a user with that name.", 404);
```

**Review by DenverCoder1 — COMMENTED** [review]
This looks great! This is a very helpful contribution.

There's still a couple suggestions earlier about the spelling of the parameter name.

Also, on [line 21 of index.php](https://github.com/komen205/github-readme-streak-stats/blob/5699b1173b207c008534817a8432d9e3eff0650b/src/index.php#L21), maybe a 500 could be sent when the token is missing from the configuration.

```diff
if (!isset($_SERVER["TOKEN"])) {
    $message = file_exists(dirname(__DIR__ . '../.env', 1))
        ? "Missing token in config. Check Contributing.md for details."
        : ".env was not found. Check Contributing.md for details.";
-    renderOutput($message);
+    renderOutput($message, 500);
}

**Review by DenverCoder1 — APPROVED** [review]
Looks great! Thanks for the contribution :tada:

**Pull request merged**